magma_access_gateway_installer/__init__.py

```diff
@@ -149,20 +142,78 @@ def validate_arbitrary_dns(args: argparse.Namespace):
+
+
+def validate_addressing(sgi_ip_address: str, sgi_gw_ip_address: str):
+    if sgi_ip_address and not sgi_gw_ip_address:
+        raise ArgumentError(
+            f"Gateway address is missing for given SGi address ({sgi_ip_address}). "
+            "Specify gateway address using --gw-ipv4-address or --gw-ipv6-address."
+        )
+    elif not sgi_ip_address and sgi_gw_ip_address:
+        raise ArgumentError(
+            f"SGi address is missing for given gateway address ({sgi_gw_ip_address}). "
+            "Specify SGi address using --ipv4-address or --ipv6-address."
+        )
+    elif sgi_ip_address and sgi_gw_ip_address:
         try:
-            ip_network(args.ip_address, False)
+            ip_network(sgi_ip_address, False)
         except ValueError:
-            raise ArgumentError(
-                f"Invalid IP address provided for --ip-address ({args.ip_address})."
-            )
+            raise ArgumentError(f"Invalid SGi IP address provided ({sgi_ip_address}).")
         try:
-            ip_address(args.gw_ip_address)
+            ip_address(sgi_gw_ip_address)
         except ValueError:
-            raise ArgumentError(
-                f"Invalid IP address provided for --gw-ip-address ({args.gw_ip_address})."
-            )
+            raise ArgumentError(f"Invalid SGi gateway IP address provided ({sgi_gw_ip_address}).")
+
+
+def configure_network(args: argparse.Namespace):
+    network_config = generate_network_config(args)
+    network_configurator = AGWInstallerNetworkConfigurator(network_config)
+    network_configurator.configure_dns()
+    network_configurator.configure_network_interfaces()
+    if any([args.ipv4_address, args.gw_ipv4_address, args.ipv6_address, args.gw_ipv6_address]):
+        installation_service_creator = AGWInstallerInstallationServiceCreator()
+        installation_service_creator.create_magma_agw_installation_service()
+        installation_service_creator.create_magma_agw_installation_service_link()
+        logger.info(
+            "Rebooting system to apply new network configuration...\n"
+            "\t\tMagma AGW installation process will be resumed automatically once machine is "
+            "back up."
+        )
+        time.sleep(5)
+        os.system("reboot")
+    else:
+        network_configurator.apply_netplan_configuration()
+
+
+def generate_network_config(args: argparse.Namespace):
+    return {
+        "sgi_ipv4_address": args.ipv4_address,
+        "sgi_ipv4_gateway": args.gw_ipv4_address,
+        "sgi_ipv6_address": args.ipv6_address,
+        "sgi_ipv6_gateway": args.gw_ipv6_address,
+        "sgi_mac_address": get_mac_address(args.sgi),
+        "s1_mac_address": get_mac_address(args.s1),
+        "dns_address": " ".join(args.dns),
+    }
+
+
+def get_mac_address(interface_name: str):
+    return netifaces.ifaddresses(interface_name)[netifaces.AF_LINK][0]["addr"]
+
+
+def set_default_sgi_interface():
```

Comment:
This should probably be a property and be called `default_sgi_interface`

Reply:
There are no classes here and, unless I'm wrong, I don't think I can use properties outside a class.